### PR TITLE
feat: Add onboarding flow UI

### PR DESCRIPTION
**Pull request by adarsh-rs**
Closes: #374

### CODE REVIEW

File: ssr/style/input.css [hunk 1]
```diff
@@ -16,4 +16,20 @@
...
+}
+@media (min-height:768px) and (max-height:819px){
+    .hot-left-arrow{
+        margin-top: 166px !important;
+    }
+}
+@media (max-height:767px){
+    .hot-left-arrow{
+        margin-top: 75px !important;
+    }
```
Comment (rupansh) [inline ssr/style/input.css:34]: Does this not work?
```
mt-20 md:mt-40 lg:mt-52 
```
any specific reason why we need these specific values for screen heights `820 to 899`?

File: ssr/src/page/post_view/video_loader.rs [hunk 2]
```diff
@@ -43,6 +54,15 @@ pub fn BgView(
         Some(())
     });
 
+    create_effect(move |_| {
+        if is_eligible_onboarding_post.get() && (!is_onboarded.get()) {
+            set_show_onboarding_popup.update(|show| *show = true);
+        } else {
+            set_show_onboarding_popup.update(|show| *show = false);
+        }
+        Some(())
```
Comment (rupansh) [inline ssr/src/page/post_view/video_loader.rs:63]: nit: do you need this line?

File: ssr/src/page/post_view/overlay.rs [hunk 3]
```diff
@@ -112,7 +112,10 @@ fn LikeAndAuthCanLoader(post: PostDetails) -> impl IntoView {
 }
 
 #[component]
-pub fn VideoDetailsOverlay(post: PostDetails) -> impl IntoView {
+pub fn VideoDetailsOverlay(
+    post: PostDetails,
+    set_eligible_onboarding_post: Option<WriteSignal<bool>>,
```
Comment (rupansh) [inline ssr/src/page/post_view/overlay.rs:117]: use context instead of prop drilling like this (Hint: PostViewCtx)

**Review by rupansh — CHANGES_REQUESTED** [review]

File: ssr/style/input.css (hunk 1 above)
Comment (adarsh-rs) [reply ssr/style/input.css:34]: I think sm, md.. breakpoints are on screen widths only.  Used these breakpoints in the screen heights for handing arrow's position.

File: ssr/src/page/post_view/video_loader.rs (hunk 2 above)
Comment (adarsh-rs) [reply ssr/src/page/post_view/video_loader.rs:63]: Yes it is required , When user gets onboarded on the app either by skipping or completing tutorial, i'm changing the local storage to true, and then updating set_show_onboarding_popup here accordingly.

File: ssr/src/page/post_view/overlay.rs (hunk 3 above)
Comment (adarsh-rs) [reply ssr/src/page/post_view/overlay.rs]: Done

File: ssr/src/page/post_view/video_loader.rs (hunk 2 above)
Comment (rupansh-gob) [reply ssr/src/page/post_view/video_loader.rs:65]: I am talking about line 63 specificially

Comment (adarsh-rs) [reply ssr/src/page/post_view/video_loader.rs]: Nope, cleaned it  in the last commit.

**Review by rupansh-gob — APPROVED** [review]

**Pull request merged**